src/Systems/MultiplayerSystem.h

```diff
@@ -79,7 +94,15 @@ class MultiplayerSystem : public System
     void setup(const std::string_view& ip, const std::string_view& port) noexcept;
     void setPlayer(const uint32_t id, const Entity entity);
     void setRoom(const glm::ivec2& room) noexcept;
+    void clearRemoteDungeonUpdates();
     void update(const float deltaTime);
+    void updateMap(const std::map<Entity, sf::Vector2<float>>& enemies,
+                   const std::map<Entity, sf::Vector2<int>>& players);
+    void sendMapDimensions(const std::unordered_map<Entity, ObstacleData>& obstacles);
+    void gatherEnemyAndPlayerPositions();
+    void sendSpawnerPosition(comm::StateUpdate& stateUpdate,
```

Comment:
czy jeden multiplayer system nie ma czasem za dużo funkcjonalności przez co encja nie ma niepotrzebnie wiele danych?

Reply:
System iteruje tylko po encjach z SynchronisedEventem, który ma (powinien mieć) pola opcjonalne w zależności od wariantu aktualizacji więc nie ma to dużego narzutu. 
Alternatywnie można by podzielić cały system na część od zmian ruchu i część od zmian stanu ale imo to tylko komplikuje sprawę.